python/mqt/core/fomac.pyi

```diff
@@ -34,6 +34,66 @@ class ProgramFormat(Enum):
     CUSTOM4 = ...
     CUSTOM5 = ...
 
+class Session:
+    """A FoMaC session for managing QDMI devices.
+
+    Allows creating isolated sessions with independent authentication settings.
+    All authentication parameters are optional and can be provided as keyword
+    arguments to the constructor.
+    """
+
+    def __init__(
+        self,
+        *,
+        token: str | None = None,
+        auth_file: str | None = None,
+        auth_url: str | None = None,
+        username: str | None = None,
+        password: str | None = None,
+        project_id: str | None = None,
+    ) -> None:
+        """Create a new FoMaC session with optional authentication.
+
+        Args:
+            token: Authentication token
+            auth_file: Path to file containing authentication information
+            auth_url: URL to authentication server
+            username: Username for authentication
+            password: Password for authentication
+            project_id: Project ID for session
+
+        Raises:
+            RuntimeError: If auth_file does not exist
+            RuntimeError: If auth_url has invalid format
+
+        Example:
+            >>> from mqt.core.fomac import Session
+            >>> # Session without authentication
+            >>> session = Session()
+            >>> devices = session.get_devices()
+            >>>
+            >>> # Session with token authentication
+            >>> session = Session(token="my_secret_token")
+            >>> devices = session.get_devices()
+            >>>
+            >>> # Session with file-based authentication
+            >>> session = Session(auth_file="/path/to/auth.json")
+            >>> devices = session.get_devices()
+            >>>
+            >>> # Session with multiple parameters
+            >>> session = Session(
+            ...     auth_url="https://auth.example.com", username="user", password="pass", project_id="project-123"
+            ... )
+            >>> devices = session.get_devices()
+        """
```

Comment:
Not an expert here, but shouldn't code examples be in Markdown fences (```)?

Reply:
The `>>>` syntax is preferred by many for its proper integration with `doctest`. In a nutshell, `doctest` can execute this as if it were plain Python code, whereas Markdown fences get ignored as static text. It may not be a significant consideration for this project, but I'd be happy to accommodate your preference here.